### ISSUE TITLE
ptr_arg wrongly suggested for &String when using clone_from 

### ISSUE THREAD

**Issue by SUPERCILEX** [rust-lang/rust-clippy#8361]
### Lint Name

ptr_arg

### Reproducer

```rust
fn foo(bar: &String) {
    let mut s = "blah blah".to_string();
    s.clone_from(bar);
}
```

### Version

```text
rustc 1.60.0-nightly (1bd4fdc94 2022-01-12)
binary: rustc
commit-hash: 1bd4fdc943513e1004f498bbf289279c9784fc6f
commit-date: 2022-01-12
host: x86_64-unknown-linux-gnu
release: 1.60.0-nightly
LLVM version: 13.0.0
```

**Comment by Jarcho** [comment]
Fixed in #8271

**Comment by SUPERCILEX** [comment]
Sweet!
